buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,82 @@
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import java.nio.file.Path;
+import org.gradle.api.Task;
+
+public class ZipPublish implements Plugin<Project> {
+    private Project project;
+
+    public final static String EXTENSION_NAME = "zipmavensettings";
+    public final static String PUBLICATION_NAME = "mavenzip";
+    public final static String STAGING_REPO = "zipstaging";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileFor"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "Publication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public String zipDistributionLocation = "/build/distributions/";
+
+    private void configMaven(Project project) {
+        final Path buildDirectory = project.getRootDir().toPath();
+        project.getPluginManager().apply(MavenPublishPlugin.class);
+        project.getExtensions().configure(PublishingExtension.class, publishing -> {
+            publishing.repositories(repositories -> {
+                repositories.maven(maven -> {
+                    maven.setName(STAGING_REPO);
+                    maven.setUrl(buildDirectory.toString() + LOCAL_STAGING_REPO_PATH);
+                });
+            });
+            publishing.publications(publications -> {
+                publications.create(PUBLICATION_NAME, MavenPublication.class, mavenZip -> {
+                    ZipPublishExtension extset = project.getExtensions().findByType(ZipPublishExtension.class);
+                    // Getting the Zip group from created extension
+                    String zipGroup = extset.getZipGroup();
+                    String zipArtifact = project.getName();
+                    String zipVersion = ZipPublishUtil.getProperty("version", project);
+                    mavenZip.artifact(project.getTasks().named("bundlePlugin"));
+                    mavenZip.setGroupId(zipGroup);
+                    mavenZip.setArtifactId(zipArtifact);
+                    mavenZip.setVersion(zipVersion);
+                });
+            });
+        });
+    }
+
+    @Override
+    public void apply(Project project) {
+        this.project = project;
+        project.getExtensions().create(EXTENSION_NAME, ZipPublishExtension.class);
+        project.afterEvaluate(evaluatedProject -> { configMaven(project); });
+        Task compileJava = project.getTasks().findByName("compileJava");
+        if (compileJava != null) {
+            compileJava.setEnabled(false);
+        }
+        Task sourceJarTask = project.getTasks().findByName("sourcesJar");
+        if (sourceJarTask != null) {
+            sourceJarTask.setEnabled(false);
+        }
+        Task javaDocJarTask = project.getTasks().findByName("javadocJar");
+        if (javaDocJarTask != null) {
+            javaDocJarTask.setEnabled(false);
+        }
```

Comment:
This will conflict when multiple publication tasks are run. Care to explain what breaks if, for example, `sourceJarTask.setEnabled(false);` is not here?

Reply:
I mean, anyhow I see we are not publishing the generated sourcesJar and javadocJar to actual nexus maven repo, this [publish](https://github.com/opensearch-project/opensearch-build/blob/main/publish/publish-snapshot.sh#L118-L119) code will ignore these jars before uploading them to nexus, so ideally these just exists on the local staging repo (filesystem) and not in maven repo.
We can also proceed to remove the task suspension from plugin code which will cause no harm but along with zips, javadocJar and sourcesJar are also included to local staging repo, and off course we stop them from publishing to nexus with the existing publish [code](https://github.com/opensearch-project/opensearch-build/blob/main/publish/publish-snapshot.sh#L118-L119) .
The other way is remove the task suspension and allow users to declare the suspension at their own will in their gradle project globally in settings .gradle as `startParameter.excludedTaskNames=["sourcesJar", "javadocJar"]` or from task using `./gradlew publishMavenzipPublicationToZipstagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER -x sourcesJar -x javadocJar`
So there are multiple options go with.

Reply:
@prudhvigodithi 

> I mean, anyhow I see we are not publishing the generated sourcesJar and javadocJar to actual nexus maven repo ...

For plugins we do [1] (and for everything else as well), probably we only skip that for snapshots:

[mapper-size-1.3.1-javadoc.jar](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar)                     2022-03-30 20:38    414035      
[mapper-size-1.3.1-javadoc.jar.asc](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar.asc)                 2022-03-30 20:38       287      
[mapper-size-1.3.1-javadoc.jar.md5](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar.md5)                 2022-03-30 20:38        32      
[mapper-size-1.3.1-javadoc.jar.sha1](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar.sha1)                2022-03-30 20:38        40      
[mapper-size-1.3.1-javadoc.jar.sha256](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar.sha256)              2022-03-30 20:38        64      
[mapper-size-1.3.1-javadoc.jar.sha512](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-javadoc.jar.sha512)              2022-03-30 20:38       128      
[mapper-size-1.3.1-sources.jar](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar)                     2022-03-30 20:38      8245      
[mapper-size-1.3.1-sources.jar.asc](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar.asc)                 2022-03-30 20:38       287      
[mapper-size-1.3.1-sources.jar.md5](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar.md5)                 2022-03-30 20:38        32      
[mapper-size-1.3.1-sources.jar.sha1](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar.sha1)                2022-03-30 20:38        40      
[mapper-size-1.3.1-sources.jar.sha256](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar.sha256)              2022-03-30 20:38        64      
[mapper-size-1.3.1-sources.jar.sha512](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1-sources.jar.sha512)              2022-03-30 20:38       128      
[mapper-size-1.3.1.jar](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar)                             2022-03-30 20:38     10416      
[mapper-size-1.3.1.jar.asc](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar.asc)                         2022-03-30 20:38       287      
[mapper-size-1.3.1.jar.md5](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar.md5)                         2022-03-30 20:38        32      
[mapper-size-1.3.1.jar.sha1](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar.sha1)                        2022-03-30 20:38        40      
[mapper-size-1.3.1.jar.sha256](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar.sha256)                      2022-03-30 20:38        64      
[mapper-size-1.3.1.jar.sha512](https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/mapper-size-1.3.1.jar.sha512)                      2022-03-30 20:38       128

We should not disable that.

[1] https://repo1.maven.org/maven2/org/opensearch/plugin/mapper-size/1.3.1/

Reply:
Cool, thanks @reta, this info really helps, I can see them on releases repo as well example: https://aws.oss.sonatype.org/content/repositories/releases/org/opensearch/opensearch-job-scheduler/1.3.0.0/.

Reply:
So based on this @reta and @dblock we should be going back to my previous changes to actually modify the existing PublishPlugin to avoid publication artifact's of sourcesJar and javadocJar if the publication is for zip.

To refer here are the changes we need to add to existing PublishPlugin, to make the sourcesJar and javadocJar are still enabled for other publications but not zip, changes are 

https://github.com/prudhvigodithi/OpenSearch/blob/gradleplugin-2.3/buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java#L124-L145

Reply:
I have made the above change for PublishPlugin and tested job-scheduler, anomaly-detection and alerting, looks good just published the zip and not the sourcesJar and javadocJar along with zips, but they are added for jar maven publication, so no interruption for existing jar publications.

Reply:
I don't see this change here, am I missing something?